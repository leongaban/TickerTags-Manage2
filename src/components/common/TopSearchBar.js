import React from 'react'

class TopSearchBar extends React.Component {
	render() {
		return (
			<div className="top-search-bar">
				<section className="fl">
					<div className="search-tickers fl">
						<input type="text" placeholder="Search Tickers"/>
						<button className="btn-blue rounded-right">Search</button>
					</div>
					<div className="add-tickers fl ml0">
						<button className="btn-blue">Add Tickers</button>
					</div>
				</section>
				<section className="fl">
					<div className="search-tags fl">
						<input type="text" placeholder="Search Tags"/>
						<button className="btn-blue rounded-right">Search</button>
					</div>
					<div className="add-tag fl ml0">
						<button className="btn-blue">Add Tag</button>
					</div>
				</section>
				<section className="username-top">
					<span>Willkommen leongaban!</span>
					<div className="user-btn">
						<p>Logout</p>
						<div className="icon-logout fr"></div>
					</div>
					<p>|</p>
					<div className="user-btn">
						<p>Settings</p>
						<div className="icon-cog fr"></div>
					</div>
				</section>
			</div>
		)
	}
}

export default TopSearchBar;