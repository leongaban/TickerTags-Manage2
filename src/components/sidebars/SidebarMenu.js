import React from 'react'
import PropTypes from 'prop-types'
export default function SidebarMenu(props) {
	return (
		<section id="sidebar-left">
			<nav>
				<ul>
					<li>
						<div className="icon">
							<div className="icon-text-height ico"></div>
						</div>
						<a href="#">My Tickers</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-list-alt ico"></div>
						</div>
						<a href="#">Analytic Reports</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-attention ico"></div>
						</div>
						<a href="#">Tag Reports</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-sort-name-up ico"></div>
						</div>
						<a href="#">Tickers Review</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-tags ico"></div>
						</div>
						<a href="#">Bulk Add Tags</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-tags-1 ico"></div>
						</div>
						<a href="#">Bulk Event Tags</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-users ico"></div>
						</div>
						<a href="#">Users</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-code ico"></div>
						</div>
						<a href="#">Rest API</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-pencil ico"></div>
						</div>
						<a href="#">Topic Annotation</a>
					</li>
					<li>
						<div className="icon">
							<div className="icon-upload-cloud ico"></div>
						</div>
						<a href="#">Deploy Blog</a>
					</li>
				</ul>
			</nav>
		</section>
	)
}