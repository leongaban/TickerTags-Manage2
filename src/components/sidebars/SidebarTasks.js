import React from 'react'
import PropTypes from 'prop-types'
import TagReportsColumn from './TagReportsColumn'
import TickersReviewColumn from './TickersReviewColumn'

export default function SidebarTasks(props) {
return (
	<section id="sidebar-right">
		<TagReportsColumn />
		<TickersReviewColumn />
	</section>
	)
}