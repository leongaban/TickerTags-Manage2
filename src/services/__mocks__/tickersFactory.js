// http://facebook.github.io/jest/docs/en/tutorial-async.html#content
import * as api from '../api'

const tickers = [
    {
        active: 1,
        company_name: "TickerTags Test",
        exchange: "",
        formatted_date: "March 20, 2017",
        industry: "",
        library_update_epoch: 1490008440,
        longname: "TickerTags Test",
        sector: "",
        ticker: "$TESTING",
        trading: 0,
        user_id: 133,
        username: "leongaban"
    }
]

export const getMyTickers = ({ user, start, limit, order }) => {
    const parameters = [];
    let url = api.tickersManage;

    if (start) { parameters.push(`start=${start}`); }
    if (limit) { parameters.push(`limit=${limit}`); }
    if (order) { parameters.push(`order=${order}`); }
    if (user.id) { parameters.push(`user_id=${user.id}`); }
    if (parameters.length > 0) { url += `?${parameters}.join('&');` }

    // return api.getTickersManage(`?${parameters}`)
    //     .then((myTickers) => myTickers);

    return new Promise((resolve, reject) => {
        process.nextTick(
            user.id === 133 ? resolve(tickers) : reject({
                error: `User with id ${user.id} not found.`
            })
        )
    });
};