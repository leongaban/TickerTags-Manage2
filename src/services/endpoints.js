export default function endpoints() {
	return {
		"login": 'app/api/login',
		"tickersManage": 'app/api/tickers/manage'
	}
};