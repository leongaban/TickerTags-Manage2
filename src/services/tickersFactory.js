import * as api from './api'

export const getMyTickers = ({ user, start, limit, order }) => {
    const parameters = [];
    let url = api.tickersManage;

    if (start) { parameters.push(`start=${start}`); }
    if (limit) { parameters.push(`limit=${limit}`); }
    if (order) { parameters.push(`order=${order}`); }
    if (user.id) { parameters.push('user_id=' + user.id); }
    if (parameters.length > 0) { url += `?${parameters}.join('&');` }

    return api.getTickersManage(`?${parameters}`)
        .then((myTickers) => myTickers);
};