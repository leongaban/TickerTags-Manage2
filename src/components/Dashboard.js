import React from 'react'
import { getCookie } from '../utils/session'
import { getMyTickers } from '../services/tickersFactory'
import TopSearchBar from './common/TopSearchBar'
import SidebarMenu from './sidebars/SidebarMenu'
import SidebarTasks from './sidebars/SidebarTasks'
import TickersTables from './tables/TickersTables'

const milliseconds = (new Date).getTime();

const userModel = {
	id: 133,
	username: 'leongaban',
	role: 'Admin'
};

const tickersFetchParams = {
	user: {},
    type: 'mine',
    tickers: [],
    service_start: 0,
    service_order: 'ticker',
    total: 0,
    date: milliseconds
};

export default function Dashboard(props) {
	let tickers = [];
	tickersFetchParams.user = userModel;

	getMyTickers(tickersFetchParams).then((myTickers) => {
		tickers = myTickers;
	});

	return (
		<div className="h100">
			<TopSearchBar />
			<section id="manage-dashboard">
				<SidebarMenu />
				<TickersTables tickers={ tickers }/>
				<SidebarTasks />
			</section>
		</div>
	)
}