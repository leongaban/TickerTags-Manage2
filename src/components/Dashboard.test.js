import React from 'react'
import { shallow } from 'enzyme'
import { getMyTickers } from '../services/tickersFactory'
jest.mock('../services/tickersFactory');
import toJson from 'enzyme-to-json'
import Dashboard from './Dashboard'
import TopSearchBar from './common/TopSearchBar'
import SidebarMenu from './sidebars/SidebarMenu'
import SidebarTasks from './sidebars/SidebarTasks'

const dashboardComponent = shallow(<Dashboard />);
const milliseconds = (new Date).getTime();

const userModel = {
	id: 133,
	username: 'leongaban',
	role: 'Admin'
};

const tickersFetchParams = {
	user: userModel,
    type: 'mine',
    tickers: [],
    service_start: 0,
    service_order: 'ticker',
    total: 0,
    date: milliseconds
};

const testing = {
    active: 1,
    company_name: "TickerTags Test",
    exchange: "",
    formatted_date: "March 20, 2017",
    industry: "",
    library_update_epoch: 1490008440,
    longname: "TickerTags Test",
    sector: "",
    ticker: "$TESTING",
    trading: 0,
    user_id: 133,
    username: "leongaban"
};

describe('<Dashboard /> component', () => {
	it('should render', () => {
		const tree = toJson(dashboardComponent);
		expect(tree).toMatchSnapshot();
	});

	it('should contain the TopSearchBar', () => {
		expect(dashboardComponent.contains(<TopSearchBar />)).toBe(true);
	});

	it('should contain the SidebarMenu', () => {
		expect(dashboardComponent.contains(<SidebarMenu />)).toBe(true);
	});

	it('should contain the SidebarTasks', () => {
		expect(dashboardComponent.contains(<SidebarTasks />)).toBe(true);
	});

	it("should make a request for and get back a list of the user's tickers", () => {
		// expect.assertions(1);
		return getMyTickers(tickersFetchParams).then(tickers => {
			expect(tickers[0]).toEqual(testing);
		});
	});
});