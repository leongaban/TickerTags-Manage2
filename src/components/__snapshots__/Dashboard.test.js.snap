// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> component should render 1`] = `
<div
  className="h100"
>
  <TopSearchBar />
  <section
    id="manage-dashboard"
  >
    <SidebarMenu />
    <TickersTables
      tickers={Array []}
    />
    <SidebarTasks />
  </section>
</div>
`;
